File: pms-client/src/main/models/task-dependency.ts
import {DependencyType} from "./dependency-type.enum";
import {Task} from './task';

export class TaskDependency {
  private taskDependency_id: number;
  private sourceTask: Task;
  private destinationTask: Task;
  private type: DependencyType;

  constructor(taskDependency_id: number, sourceTask: Task, destinationTask: Task, type: DependencyType) {
    this.taskDependency_id = taskDependency_id;
    this.sourceTask = sourceTask;
    this.destinationTask = destinationTask;
    this.type = type;
  }

  getTaskDependency_id(): number {
    return this.taskDependency_id;
  }

  setTaskDependency_id(value: number) {
    this.taskDependency_id = value;
  }

  getSourceTask(): Task {
    return this.sourceTask;
  }

  setSourceTask(value: Task) {
    this.sourceTask = value;
  }

  getDestinationTask(): Task {
    return this.destinationTask;
  }

  setDestinationTask(value: Task) {
    this.destinationTask = value;
  }

  getType(): DependencyType {
    return this.type;
  }

  setType(value: DependencyType) {
    this.type = value;
  }
}

File: pms-core/pms-core-ejb/src/main/java/tn/rnu/eniso/pms/core/ejb/entities/TaskDependency.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.rnu.eniso.pms.core.ejb.entities;

import java.io.Serializable;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.Table;

/**
 *
 * @author bacali
 */
@Entity
@Table(name = "TBL_TASK_DEPENDENCY")
public class TaskDependency implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    private Task sourceTask;
    @OneToOne
    private Task destinationTask;
    private DependencyType type;
    
    public TaskDependency() {
    }

   
    
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Task getSourceTask() {
        return sourceTask;
    }

    public void setSourceTask(Task sourceTask) {
        this.sourceTask = sourceTask;
    }

    public Task getDestinationTask() {
        return destinationTask;
    }

    public void setDestinationTask(Task destinationTask) {
        this.destinationTask = destinationTask;
    }

    public DependencyType getType() {
        return type;
    }

    public void setType(DependencyType type) {
        this.type = type;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (id != null ? id.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof TaskDependency)) {
            return false;
        }
        TaskDependency other = (TaskDependency) object;
        if ((this.id == null && other.id != null) || (this.id != null && !this.id.equals(other.id))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "tn.rnu.eniso.pms.core.ejb.entities.TaskDependency[ id=" + id + " ]";
    }
    
}
